src/Bicep.Core/Emit/EmitLimitationCalculator.cs

```diff
@@ -93,5 +96,153 @@ public static EmitLimitationInfo Calculate(SemanticModel semanticModel)
+                        ModulePropertyNameSyntax = namePropSyntax
+                    };
+                    return r;
+                })
+                .Where(x => x.ModulePropertyNameSyntax is not null && x.ModulePropertyNameValue is not null)
+                .GroupBy(x => (x.ModulePropertyScopeType, x.ModulePropertyScopeValue, x.ModulePropertyNameValue!), ModuleComparer.Instance)
+                .Where(group => group.Count() > 1);
+
+            foreach (var duplicatedModuleGroup in duplicateModules)
+            {
+                var duplicatedModuleNames = string.Join(", ", duplicatedModuleGroup.Select(x => x.ModuleName));
+                foreach (var duplicatedModule in duplicatedModuleGroup)
+                {
+                    yield return DiagnosticBuilder.ForPosition(duplicatedModule.ModulePropertyNameSyntax!).ModuleMultipleDeclarations(duplicatedModuleNames);
+                }
+            }
+        }
+
+
+        // comparers below are very simple now, however in future it might be used to do more exact comparsion on property value to include interpolations
+        // also, we expect StringSyntax as values it can be other types as well (function calls, variable accesses, etc.)
+        private class ResourceComparer : IEqualityComparer<(string ResourceTypeFQDN, ResourceSymbol? ResourceScope, StringSyntax PropertyNameValue)>
+        {
+            public static readonly ResourceComparer Instance = new();
+
+            public bool Equals((string ResourceTypeFQDN, ResourceSymbol? ResourceScope, StringSyntax PropertyNameValue) x, (string ResourceTypeFQDN, ResourceSymbol? ResourceScope, StringSyntax PropertyNameValue) y)
+            {
+
+                if (!string.Equals(x.ResourceTypeFQDN, y.ResourceTypeFQDN, StringComparison.OrdinalIgnoreCase))
+                {
+                    return false;
+                }
+                
+                if (x.ResourceScope != y.ResourceScope)
+                {
+                    return false;
+                }
+
+                var xv = x.PropertyNameValue.TryGetLiteralValue();
+                var yv = y.PropertyNameValue.TryGetLiteralValue();
+
+                //if literal value is null, we assume resources are not equal, as this indicates that interpolated value is used
+                //and as for now we're unable to determine if they will have equal values or not.
+
+                return xv is not null && yv is not null && string.Equals(xv, yv, StringComparison.OrdinalIgnoreCase);
+            }
+
+            public int GetHashCode((string ResourceTypeFQDN, ResourceSymbol? ResourceScope,  StringSyntax PropertyNameValue) obj)
+            {
+                return HashCode.Combine(obj.ResourceTypeFQDN.ToLowerInvariant(), obj.ResourceScope, obj.PropertyNameValue.TryGetLiteralValue()?.ToLowerInvariant());
```

Comment:
I've never benchmarked this, but I'd assume that using `StringComparer.OrdinalIgnoreCase.GetHashCode(<string>)` would be more performant than `<string>.ToLowerInvariant()` as it avoids allocation - e.g.:

```csharp
return HashCode.Combine(
    StringComparer.OrdinalIgnoreCase.GetHashCode(obj.ResourceTypeFQDN),
    obj.ResourceScope,
    StringComparer.OrdinalIgnoreCase.GetHashCode(obj.PropertyNameValue.TryGetLiteralValue() ?? ""));

Reply:
That might be true, didn't check that either. Of course in a result we will be building hash from values' hashes but that should be fine, as this is only for initial equality detection. Will change that.

Reply:
> building hash from values' hashes

Yeah I've actually seen very little info on how `HashCode.Combine()` works, but I'd hope that it's optimized for handling int32 values - as `[int].GetHashCode()` is effectively a no-op.

Usually I do something like `GetHashCode(..) => val1.GetHashCode() ^ val2.GetHashCode() ^ val3.GetHashCode();`, but I suspect `HashCode.Combine()` is probably optimized to give a better distribution of hash values than using XOR.

Reply:
I used `new HashCode()` to build hash codes as described https://docs.microsoft.com/en-us/dotnet/api/system.hashcode?view=dotnet-plat-ext-5.0#instance-methods